### PR TITLE
Update k6 dependency to latest stable v0.40.0

### PR DESCRIPTION
**Pull request by imiric**

### CODE REVIEW

**Comment by imiric** [comment]
@inancgumus You shouldn't approve this if the tests are failing for a legitimate reason. :sweat_smile:

See https://loadimpact.slack.com/archives/C02H6RF6X1P/p1662643395772169 and https://github.com/grafana/k6/commit/d69cb582741618b93d8321e021009aa375761250.

Looking into it now, but I'll have to leave early today, so if I don't finish maybe someone else can take a look? Otherwise I'll continue tomorrow.

**Comment by inancgumus** [comment]
Hmm, the new k6 version has [some complaints](https://github.com/grafana/xk6-browser/runs/8250141304?check_suite_focus=true).

More info here: https://github.com/grafana/k6/pull/2602

**Comment by inancgumus** [comment]
> @inancgumus You shouldn't approve this if the tests fail for a legitimate reason. 😅

Yeah, my bad; I became aware of the issue after I approved the PR. It was looking harmless :)

**Comment by imiric** [comment]
I pushed a WIP attempt at fixing the test issue in ccf39b4, but some tests fail with a nil panic in `chromium.NewBrowserType()` while looking up the `Registry` at `vu.InitEnv().Registry`. According to @mstoykov the `modulestest` changes better replicate real k6 behavior, but I'm not seeing a way to make it work, and have to leave soon. `Runtime.MoveToVUContext()` should be used instead of setting `Runtime.VU.StateField` directly, but either way it results in a nil panic.

I'll look more into this tomorrow, but would appreciate if someone else could take a look as well. Mihail would be able to help from Monday.

Upgrading to k6 v0.40.0 is not a blocker for releasing xk6-browser v0.5.0, so we could release with k6 v0.39.0 as well. Or we could delay the release by a day, and fix this on Monday.

**Comment by imiric** [comment]
I hackfixed the nil panic and modulestest integrity check, but now the logger is racy... :sob:

Looking into it...

File: tests/test_browser.go
Comment (inancgumus) [inline tests/test_browser.go:137]: I have a funny feeling about this fix :flushed: Should we better work on a proper fix in the next cycle? 🤔

**Comment by imiric** [comment]
OK, so it should be fixed with aaae86c. Like I mentioned, it's a _hackfix_ and we should find a cleaner way of doing this.

@mstoykov Could you also review aaae86c and let us know if there's a better way? :pray:

File: tests/test_browser.go
Comment (imiric) [reply tests/test_browser.go:137]: Yes, this is temporary and we should find a better way, hopefully with @mstoykov's help.

File: k6ext/k6test/vu.go
Comment (inancgumus) [inline k6ext/k6test/vu.go:26]: I would argue that k6 APIs (especially the new ones like `modulestest`) are in flux, and we better keep the wrapper and other helpers. Also, the helpers provide easy-to-remember (at least, for me) names for the common and mostly verbose operations.

**Review by inancgumus — COMMENTED** [review]
🤔

File: tests/test_browser.go
Comment (inancgumus) [reply tests/test_browser.go:137]: 🤞

**Comment by imiric** [comment]
OK, merging this then. If someone has an improvement suggestion, we can work on it for the next release.

**Review by ankur22 — APPROVED** [review]

File: tests/test_browser.go
Comment (mstoykov) [reply tests/test_browser.go:137]: Something like https://github.com/grafana/xk6-browser/pull/525 will is how the API is more appropriate to be used.

File: k6ext/k6test/vu.go
Comment (mstoykov) [reply k6ext/k6test/vu.go:26]: 1. I would argue you should not have `WaitOnRegistered` *before* the `LoopStart` you might want to have it after to wait on everything to finish in case of errors. But adding it before just masks that a previous test didn't do something properly and you are reusing the same loop.
2. You can still just wrap `modulestest.Runtime` instead of VU ... but maybe you will need to rename the struct ;). And if you embed modulestest.Runtime (or the wrapper) in testBrowser, you should need to touch as many places to rename the access to `VU` due to how embedding fields in structs works.

Comment (inancgumus) [reply k6ext/k6test/vu.go:26]: 1. Yeah, I was suspicious of adding `WaitOnRegistered` there. The problem is in the tests (`slow_mo` tests) that share the same browser and VU.
2. I guess `modulestest.Runtime` is added in k6@v0.4.0 and we can take advantage of it 👍

**Pull request merged**
